tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1149,7 +1149,9 @@ def pad_to_bounding_box_internal(image, offset_height, offset_width,
         array_ops_stack.stack([
             0, 0, offset_height, after_padding_height, offset_width,
             after_padding_width, 0, 0
-        ]), [4, 2])
```

Comment:
This has been here since 2018.  Are you sure this is the cause of your issue?

Reply:
Can't be 100% sure because I was not able to find the code for pywrap_tfe, I just find a .so file for that.

The only thing that could affect apart from that is the context (`_ctx`), for reference, I am talking about the following code at `gen_array_ops.py`:

```
_result = pywrap_tfe.TFE_Py_FastPathExecute(
        _ctx, "Reshape", name, tensor, shape)
``` 

That said, in both scenarios when I print the context it prints the same information:
```
context: Eager TensorFlow Context with 1 devices
   Device 0: /job:localhost/replica:0/task:0/device:CPU:0
```

I don't think the fact that the code has been around since 2018 is a valid reason not to question it.

Reply:
> I don't think the fact that the code has been around since 2018 is a valid reason not to question it.

Well that, plus our CI tests all pass, and that we have several other instances in this same file (and throughout the codebase) that use lists with reshape.

How exactly are you triggering the error in the first place?

Reply:
> Well that, plus our CI tests all pass, and that we have several other instances in this same file (and throughout the codebase) that use lists with reshape.

CI and tests never cover all corner cases, if they did, there would never be bugs.


This is the minimum reproducible code I can create that produces the error, I took the values by printing the values from the environment that fails:

```
import tensorflow as tf
from tensorflow.python import pywrap_tfe as pywrap_tfe
from tensorflow.python.eager import context as _context

_ctx = _context._context or _context.context()

name = None
tensor = tf.constant([ 0 , 0,  0, 16,  0,  0,  0,  0])
shape = [4, 2]
_result = pywrap_tfe.TFE_Py_FastPathExecute(
        _ctx, "Reshape", name, tensor, shape
)
```

This will produce: 
```
Traceback (most recent call last):
  File "/tmp/pose/min_repro.py", line 11, in <module>
    _result = pywrap_tfe.TFE_Py_FastPathExecute(
tensorflow.python.eager.core._FallbackException: This function does not handle the case of the path where all inputs are not already EagerTensors.
```

I got the values used as input on the code above by running this [example](https://tfhub.dev/google/lite-model/movenet/multipose/lightning/tflite/float16/1).

Reply:
I am not calling an op directly, I created the min reproducible example with what I got from the original code.

This file `python3.10/site-packages/tensorflow/python/ops/gen_array_ops.py` which I can't find on this same repo, is what is being called by the code this PR changes. This is its content of interest from that file:

```
....
Returns:
    A `Tensor`. Has the same type as `tensor`.
  """
  _ctx = _context._context or _context.context()
  tld = _ctx._thread_local_data
  if tld.is_eager:
    try:
      _result = pywrap_tfe.TFE_Py_FastPathExecute(
        _ctx, "Reshape", name, tensor, shape) <--- HERE THE SHAPE ARRIVES AS A LIST, AND THUS, IT FAILS.
      return _result
    except _core._NotOkStatusException as e:
      _ops.raise_from_not_ok_status(e, name)
    except _core._FallbackException: < ---THIS IS WHERE THE FALLBACK HAPPENS ON THE FIRST ENVIRONMENT I TRIED (the one that works)
      pass
    except Exception as e:
        print(e)
    try:
      return reshape_eager_fallback(
          tensor, shape, name=name, ctx=_ctx)
    except _core._SymbolicException:
      pass  # Add nodes to the TensorFlow graph.
...
```

Reply:
BTW all this is coming from invoking: 
```
image = tf.image.pad_to_bounding_box(image, 0, 0, target_height, target_width)
```

Reply:
The `gen_array_ops.py` is auto-generated during the build process, so you won't find it in the repo.

Yes, in eager mode, it takes the eager fallback.  That is where the `shape` is turned into a tensor for you automatically.  I've extracted the relevant `gen_array_ops.py` code [here](https://colab.research.google.com/drive/1HmndS0LoA4riimVwAruiHDPHea6h2MsZ?usp=sharing), that you can hopefully run and play around with.  So all those `gen_array_ops.reshape(..., [4, 2])` work fine without needing to explicitly convert the shape into a tensor.

Are you trying to run this: https://github.com/tensorflow/docs/blob/master/site/en/hub/tutorials/movenet.ipynb?

If you click the "Run in Google Colab" button, both the TF `"movenet_lightning"` and the tf-lite model you linked to above, `"movenet_lightning_f16.tflite"`, work for me without issue.

Reply:
So, the issue seems to be that in eager mode it is not taking the eager fallback, and fails silently. That's also the reason  
 for it to work when I add the code of this PR, because doesn't take the fallback but the input is already an Eager tensor so it can process it.

> Yes, in eager mode, it takes the eager fallback. That is where the shape is turned into a tensor for you automatically. I've extracted the relevant gen_array_ops.py code [here](https://colab.research.google.com/drive/1HmndS0LoA4riimVwAruiHDPHea6h2MsZ?usp=sharing), that you can hopefully run and play around with.

I already played with this code, and pasted above the output of some debug prints in both cases the working and the failing one, please take a look at my previous comments.

> Are you trying to run this: https://github.com/tensorflow/docs/blob/master/site/en/hub/tutorials/movenet.ipynb?
If you click the "Run in Google Colab" button, both the TF "movenet_lightning" and the tf-lite model you linked to above, "movenet_lightning_f16.tflite", work for me without issue.

I am working with [this](https://tfhub.dev/google/lite-model/movenet/multipose/lightning/tflite/float16/1). But I don't want to run it in colab. I am able to run it locally, but it just does not always work.

So, I would like to understand why executing on eager mode, it doesn't take the eager fallback and instead fails silently. That code doesn't seem to be public, seems to happen at `_pywrap_tfe.so` (`pywrap_tfe.TFE_Py_FastPathExecute`)

Reply:
I need some time to create a minimal reproducible script, the environment is kind of big right now and I don't have the error yet well localized inside the pywrap_tfe module.
Will get back in a couple days

Reply:
Hey @cantonios,
I managed to solve the issue after finding a different one. Downgrading NumPy to `1.24.3` from `1.25.2` it started to work as expected, so I'll close this. I don't really know how that is related, but found a different issue with the program producing a segmentation fault after importing NumPy, after downgrading it, everything started to work.